scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -266,6 +267,12 @@ cdef dict _unpack_dist(object dist, str dist_type, list meths,
                     # some distributions require array inputs.
                     x = np.asarray((x-self.loc)/self.scale)
                     return max(0, self.dist.dist._pdf(x, *self.args)/self.scale)
+                def logpdf(self, x):
+                    # some distributions require array inputs.
+                    x = np.asarray((x-self.loc)/self.scale)
+                    if self.pdf(x) > 0:
```

Comment:
@tirthasheshpatel I thought it is safer to apply such a check since if we use _logpdf, there is no guarantee that we get a meaningful value if x is not in the domain of the distribution. or is it not needed since the support will be inferred if dist is passed to NumericalInversePolynomial?

Reply:
I wouldn't be surprised if there are some distributions with infinite support that give zero or even negative `pdf` values within the support just due to numerical noise, so a check is not a terrible idea.

Reply:
I agree. The checks look good!